graphql_compiler/compiler/blocks.py

```diff
@@ -244,7 +251,13 @@ class Traverse(BasicBlock):
 
     __slots__ = ("direction", "edge_name", "optional", "within_optional_scope")
 
-    def __init__(self, direction, edge_name, optional=False, within_optional_scope=False):
+    def __init__(
+        self,
+        direction: str,
+        edge_name: str,
+        optional: bool = False,
+        within_optional_scope: bool = False,
+    ) -> None:
```

Comment:
Upon first read through I was confused because the documentation for this function has a `Returns` section, but the type hint says nothing is returned. Might just be a special case for `__init__` functions, but is it correct to say that this returns a new traverse object?

Also (not particularly related to the scope of this PR), the `Args` section is missing `within_optional_scope`.

Reply:
Yeah, the `Returns` on `__init__` is something we probably should eliminate everywhere, maybe as a tech debt item. More generally, it is `__new__` that returns a new (uninitialized) object, which then `__init__` initializes by mutating the `self` reference. So while a `Returns` section on `__init__` can reasonably be thought of as describing the *intent*, it is technically incorrect.

I'm happy to remove the `Returns` section here, and document `within_optional_scope`. Nice catch! I'm not sure why `pydocstyle` didn't catch that.

Reply:
I found one more instance where `within_optional_scope` wasn't described, and I removed all the `Returns` sections from `__init__` methods.